src/media/local-stream.ts

```diff
@@ -60,6 +81,65 @@ abstract class _LocalStream extends Stream {
       this[StreamEventNames.Muted].emit(isMuted);
     }
   }
+
+  /**
+   * Adds an effect to a local stream.
+   *
+   * @param name - The name of the effect.
+   * @param effect - The effect to add.
+   */
+  async addEffect(name: string, effect: TrackEffect): Promise<void> {
+    // Check that the loaded effect is the latest one and dispose if not
+    if (this.loadingEffects.indexOf(name) > -1) {
+      await effect.dispose();
+      throw new Error(`Effect "${name}" not required after loading`);
+    }
+
+    // Prevent loading multiple effects wth the same name
+    if (this.effects.findIndex((e) => e.name === name)) {
+      await effect.dispose();
+      throw new Error(`Effect already exists with name "${name}"`);
```

Comment:
According to the web client team, there's a use case to replace an earlier version of an effect with a newer one (which may have different params)--I think that's what the current webrtc-core supports.  If we want to enforce they dispose the old one first, we should check in with them.

Reply:
Ah I must have misread.  For now I can replicate their implementation.